misc/chrome_extension/content-script.js

```diff
@@ -101,42 +122,291 @@ chrome.runtime.onMessage.addListener(
 		//Gets the nodeValue for XPATH
 		//Gets the textContent for CSS
 		for (let key in metadata) {
-			if (metadata[key]['type'] == "CSS") {
-				elements.push(parseCss(key, metadata[key]['path'], true));
+			let currentExcludeObject = metadata[key];
+			if (currentExcludeObject['type'] === 'CSS') {
+				elementsToReturn.push(parseCss(key, currentExcludeObject['path'], true));
 			}
-			else if (metadata[key]['type'] == "XPATH") {
-				elements.push(parseXPath(key, metadata[key]['path'], true));
+			else if (currentExcludeObject['type'] === 'XPATH') {
+				elementsToReturn.push(parseXPath(key, currentExcludeObject['path'], true));
 			}
 		}
 
 		//Adds the elements to exclude in the elementsToHide
 		for (let key in exclude) {
-			if (exclude[key]['type'] == "CSS") {
-				elementsToHide.push(parseCss(key, exclude[key]['path'], false));
+			let currentMetadataObject = exclude[key];
+			if (currentMetadataObject['type'] === 'CSS') {
+				elementsToHide.push(parseCss(key, currentMetadataObject['path'], false));
 			}
-			else if (exclude[key]['type'] == "XPATH") {
-				elementsToHide.push(parseXPath(key, exclude[key]['path'], false));
+			else if (currentMetadataObject['type'] === 'XPATH') {
+				elementsToHide.push(parseXPath(key, currentMetadataObject['path'], false));
 			}
 		}
 
 		//Reduces opacity of elements in elementsToHide
 		//If an error is found, adds it in elements to send back
-		elementsToHide.forEach(function (elementArray) {
-			if (elementArray['type'] != "__error") {
-				elementArray['value'].forEach(function (element) {
+		elementsToHide.forEach(function (elementObject) {
+			if (elementObject['type'] != '__error') {
+				elementObject['value'].forEach(function (element) {
 					element.style.opacity = 0.1;
 				}, this);
 			}
 			else {
-				elements.push(elementArray);
+				elementsToReturn.push(elementObject);
 			}
 		}, this);
 
 		//Send back the values found
 		setTimeout(function () {
 			chrome.runtime.sendMessage({
-				return: JSON.stringify(elements)
+				return: JSON.stringify(elementsToReturn)
 			});
 		}, 1);
+	}
+
+
+	/**
+	 * Adds the __highlight class to the object at the currently being moused over
+	 * 
+	 * @param {event} event 
+	 * @returns nothing
+	 */
+	function mouseoverHandler(event) {
+
+		if (!ableToMouseOver) {
+			return;
+		}
+
+		if (event.target === document.body ||
+			(prev && prev === event.target)) {
+			return;
+		}
+
+		try {
+			if (prev) {
```

Comment:
What is `prev`, maybe you need a better name and some comments about what is going on here.